plugins/survey/src/types.ts

```diff
@@ -0,0 +1,151 @@
+// limitations under the License.
+//
+
+import { AttachedDoc, Class, CollectionSize, Doc, Markup, Ref, Space } from '@hcengineering/core'
+import { Attachment } from '@hcengineering/attachment'
+import { Person } from '@hcengineering/contact'
+import { ComponentType, SvelteComponent } from 'svelte'
+import { Resource } from '@hcengineering/platform'
+
+/**
+ * @public
+ *
+ * @link https://www.npmjs.com/package/lexorank
+ */
+export type Rank = string
+
+/**
+ * ∈ [0, 1]
+ */
+export type Fraction = number
+
+/** @public */
+export interface Survey extends Space {
+  name: string
+  questions: CollectionSize<Question>
+  requests: CollectionSize<Request>
+  // TODO: isAssessment: boolean
+}
+
+/** @public */
+export interface Question extends AttachedDoc<Survey, 'questions', Survey> {
+  title: Markup
+  rank: Rank
+  attachments: CollectionSize<Attachment>
+  assessment?: AssessmentData<this>
+}
+
+/** @public */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export interface AssessmentData<Q extends Question> {
+  weight: number
+}
+
+/** @public */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unused-vars
+export interface AnswerData<Q extends Question> {}
+
+/** @public */
+export interface QuestionOption {
+  label: string
+}
+
+/** @public */
+export interface SingleChoiceQuestion extends Question {
+  options: QuestionOption[]
+  shuffle: boolean
+}
+
+/** @public */
+export interface SingleChoiceAnswerData extends AnswerData<SingleChoiceQuestion> {
+  selection?: number
+}
+
+/** @public */
+export interface SingleChoiceAssessmentData extends AssessmentData<SingleChoiceQuestion> {
+  correctAnswer: SingleChoiceAnswerData
+}
+
+/** @public */
+export interface MultipleChoiceQuestion extends Question {
+  options: QuestionOption[]
+  shuffle: boolean
+}
+
+/** @public */
+export interface MultipleChoiceAnswerData extends AnswerData<MultipleChoiceQuestion> {
+  selections: number[]
+}
+
+/** @public */
+export interface MultipleChoiceAssessmentData extends AssessmentData<MultipleChoiceQuestion> {
+  correctAnswer: MultipleChoiceAnswerData
+}
+
+/** @public */
+export interface ReorderQuestion extends Question {
+  options: QuestionOption[]
+  shuffle: boolean
+}
+
+/** @public */
+export interface ReorderAnswerData extends AnswerData<ReorderQuestion> {
+  order: number[]
+}
+
+/** @public */
+export interface ReorderAssessmentData extends AssessmentData<ReorderQuestion> {
+  correctAnswer: ReorderAnswerData
+}
+
+/** @public */
+export interface SurveyRequest extends AttachedDoc<Survey, 'requests', Survey> {
+  // TODO: Should it be Employee, no?
+  // TODO: Should it be an array of assignees instead?
+  assignee: Ref<Person>
+
+  results: CollectionSize<SurveyResult>
+  // TODO: maxAttempts?: number
+  // TODO: dueDate?: Timestamp
+}
+
+export interface SurveyResult extends AttachedDoc<SurveyRequest, 'results', Survey> {
```

Comment:
Once submitted it might be handy to calculate and store the score for the whole result and whether the user passed (result compared agains the threshold) so we don't need to calculate it every time. Also, might then be simpler to have some filters/etc for this conditions for the results.
Is this applicable to basic surveys?